### ISSUE TITLE
NoMethodError - undefined method `-' for nil:NilClass

### ISSUE THREAD

**Issue by clausjuhl** [tabulapdf/tabula#358]
This error-message displays when trying to extract data from selected areas:

Tabula API version: 1.0.0
Filename: BrancheBeskrivelser.pdf
Internal Server Error (500)
    
      
        Request Method:
        POST
      
      
        Request URL:
      http://127.0.0.1:34555/pdf/dec03fe079f218a31b80533ff042a3295a5d6f9b/data

(NoMethodError) undefined method `-' for nil:NilClass

The file (in danish) producing the error is located here: http://www.dst.dk/pukora/epub/upload/11119/branche.pdf

**Comment by asyavuz** [comment]
I got same error with this file (in Turkish): http://dokuman.osym.gov.tr/pdfdokuman/2015/OSYS/OSYS2015YerlestirmeMinMaxTablo-423072015.pdf

**Comment by mtigas** [comment]
Hi @clausjuhl @asyavuz: what operating system are you using? I have a theory about this that might affect users on Mac.

**Comment by asyavuz** [comment]
Hi @mtigas Mac OSX Yosemite 10.10.4.

**Comment by jazzido** [comment]
Hi @asyavuz, 

Thanks for reporting the issue. What were the steps that led to the error? I successfully extracted the first 100-odd pages of your document by:

  - Uploading file
  - Clicking on 'autodetect tables'
  - Clicking on 'Preview and export extracted data'

**Comment by clausjuhl** [comment]
@mtigas. I'm using windows 7 Professional, SP1. Browser is Chrome (44.0.2403.130). The steps that led to the error was:

1. Uploaded file
2. Clicked 'autodetect tables'
3. Adjusted detected table and clicked 'Repeat this selection'
4. Clicked 'Preview'

**Comment by asyavuz** [comment]
Hi @jazzido,

The steps i followed were exactly the same as yours. But now, I tried restarting computer, removing file and re-uploading, it worked this time. Sorry for bothering you.

**Comment by jazzido** [comment]
No problem, @asyavuz. What you describe makes the issue even weirder, so all information is useful for debugging :)

**Comment by jeremybmerrill** [comment]
I suspect this issue has been fixed in 6d5c222, so the problem will be fixed in the next bugfix release of Tabula, which should be released soon. I was able to replicate @clausjuhl's error before 6d5c222, but the same steps worked correctly after 6d5c222 and with tabula-extractor master.

That fix relates to PDFs with lots of pages when users create selections that aren't rendered onto the page (with autodetect or, I presume, with repeat).
